src/utils/decorators/tooltip-decorator/tooltip-decorator.js

```diff
@@ -173,51 +173,77 @@ const TooltipDecorator = (ComposedComponent) => {
         if (this.state.isVisible) this.setState({ isVisible: false });
         return;
       }
-      const alignment = this.state.tooltipAlign || this.props.tooltipAlign || 'center';
-      const position = this.state.tooltipPosition || this.props.tooltipPosition || 'top';
+      const tooltipAlign = this.state.tooltipAlign || this.props.tooltipAlign || 'center';
+      const tooltipPosition = this.state.tooltipPosition || this.props.tooltipPosition || 'top';
       const shifts = calculatePosition(tooltip, target);
-      const isTooltipOffScreenRight = this.isTooltipOffScreenRight(position, alignment, tooltip.offsetWidth, shifts);
-      const inputDistanceOffset = this.getInputDistanceOffset(position);
+      const isTooltipOffScreenLeft = this.isTooltipOffScreenLeft(tooltipPosition, tooltipAlign, shifts);
+      const isTooltipOffScreenRight = this.isTooltipOffScreenRight(
+        tooltipPosition,
+        tooltipAlign,
+        tooltip.offsetWidth,
+        shifts
+      );
+      const inputDistanceOffset = this.getInputDistanceOffset(tooltipPosition);
+
+      if (isTooltipOffScreenLeft) {
+        this.realignOffscreenTooltip(tooltipPosition, 'left');
+        return;
+      }
 
       if (isTooltipOffScreenRight) {
-        this.realignOffscreenTooltip(position, alignment);
+        this.realignOffscreenTooltip(tooltipPosition, 'right');
+        return;
       }
 
-      if (position === 'top' || position === 'bottom') {
-        topPosition = shifts.tooltipDistances[position] + inputDistanceOffset;
-        leftPosition = shifts.vertical[alignment] + this.getHorizontalOffset();
+      if (tooltipPosition === 'top' || tooltipPosition === 'bottom') {
+        topPosition = shifts.tooltipDistances[tooltipPosition] + inputDistanceOffset;
+        leftPosition = shifts.vertical[tooltipAlign] + this.getHorizontalOffset();
       } else {
-        topPosition = shifts.horizontal[alignment];
-        leftPosition = shifts.tooltipDistances[position] + inputDistanceOffset;
+        topPosition = shifts.horizontal[tooltipAlign];
+        leftPosition = shifts.tooltipDistances[tooltipPosition] + inputDistanceOffset;
       }
 
       styleElement(tooltip, 'top', `${topPosition}px`);
       styleElement(tooltip, 'left', `${leftPosition}px`);
     };
 
-    isTooltipOffScreenRight = (position, alignment, tooltipWidth, shifts) => {
+    isTooltipOffScreenLeft = (tooltipPosition, tooltipAlign, shifts) => {
       let pointerX;
 
-      if (position === 'left') {
+      if (tooltipPosition === 'right' || tooltipAlign === 'left') {
         return false;
       }
 
-      if (position === 'right') {
-        pointerX = shifts.tooltipDistances.right;
+      if (tooltipPosition === 'left') {
+        pointerX = shifts.tooltipDistances.left;
       } else {
-        pointerX = shifts.vertical[alignment];
+        pointerX = shifts.vertical[tooltipAlign];
       }
 
-      return window.innerWidth < pointerX + tooltipWidth;
+      return pointerX < 16;
```

Comment:
Why specifically `16`?

Reply:
UX asked for the tooltip to reposition when the gap between the edge of the tooltip and its container gets to less than 16px (on both right and left). I've added it to the offscreen right calculation below as well